src/qiboml/models/abstract.py

```diff
@@ -0,0 +1,67 @@
+"""Defines the general structure of a qiboml module"""
+
+from abc import ABC, abstractmethod
+from dataclasses import dataclass
+
+from qibo import Circuit
+from qibo.config import raise_error
+from qibo.gates import abstract
+
+from qiboml.backends import TensorflowBackend as JaxBackend
+
+
+@dataclass
+class QuantumCircuitLayer(ABC):
+
+    nqubits: int
+    qubits: list[int] = None
+    circuit: Circuit = None
+    initial_state: "ndarray" = None
+    backend: "qibo.backends.Backend" = JaxBackend()
+
+    def __post_init__(self) -> None:
+        if self.qubits is None:
+            self.qubits = range(self.nqubits)
```

Comment:
Not a `list`, so it would not respect the type hint.

Either change the hint, or, simply, just turn this into a `list`:
```suggestion
            self.qubits = list(range(self.nqubits))
```
(range is definitely a more compact representation, but it will only allow for contiguous indices)